packages/google_sign_in/google_sign_in/example/ios/RunnerTests/GoogleSignInTests.m

```diff
@@ -26,62 +26,406 @@ - (void)setUp {
   [super setUp];
   self.mockBinaryMessenger = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
   self.mockPluginRegistrar = OCMProtocolMock(@protocol(FlutterPluginRegistrar));
-  self.mockSharedInstance = [OCMockObject partialMockForObject:[GIDSignIn sharedInstance]];
+
+  id mockSharedInstance = OCMClassMock([GIDSignIn class]);
+  OCMStub([mockSharedInstance sharedInstance]).andReturn(mockSharedInstance);
+  self.mockSharedInstance = mockSharedInstance;
+
   OCMStub(self.mockPluginRegistrar.messenger).andReturn(self.mockBinaryMessenger);
   self.plugin = [[FLTGoogleSignInPlugin alloc] init];
   [FLTGoogleSignInPlugin registerWithRegistrar:self.mockPluginRegistrar];
 }
 
 - (void)tearDown {
-  [((OCMockObject *)self.mockSharedInstance) stopMocking];
+  [self.mockSharedInstance stopMocking];
   [super tearDown];
 }
 
+- (void)testUnimplementedMethod {
+  FlutterMethodCall *methodCall = [FlutterMethodCall methodCallWithMethodName:@"bogus"
+                                                                    arguments:nil];
+
+  XCTestExpectation *expectation = [self expectationWithDescription:@"expect result returns true"];
+  [self.plugin handleMethodCall:methodCall
+                         result:^(id result) {
+                           XCTAssertEqualObjects(result, FlutterMethodNotImplemented);
+                           [expectation fulfill];
+                         }];
+  [self waitForExpectationsWithTimeout:5.0 handler:nil];
+}
+
+- (void)testSignOut {
+  FlutterMethodCall *methodCall = [FlutterMethodCall methodCallWithMethodName:@"signOut"
+                                                                    arguments:nil];
+
+  XCTestExpectation *expectation = [self expectationWithDescription:@"expect result returns true"];
+  [self.plugin handleMethodCall:methodCall
+                         result:^(id result) {
+                           XCTAssertNil(result);
+                           [expectation fulfill];
+                         }];
+  [self waitForExpectationsWithTimeout:5.0 handler:nil];
+  OCMVerify([self.mockSharedInstance signOut]);
+}
+
+- (void)testDisconnect {
+  FlutterMethodCall *methodCall = [FlutterMethodCall methodCallWithMethodName:@"disconnect"
+                                                                    arguments:nil];
+
+  [self.plugin handleMethodCall:methodCall
+                         result:^(id result){
+                         }];
```

Comment:
Don't we want to assert that it returned a (non-error) result?

Reply:
The tests that don't check the result are doing do because the result callback happens from a `GIDSignIn.delegate` method.  
https://github.com/flutter/plugins/blob/b548ae18646d1725a8e1f004c26bf42e4d362aa8/packages/google_sign_in/google_sign_in/ios/Classes/FLTGoogleSignInPlugin.m#L132-L135
sends the response in the `GIDSignInDelegate` call.
https://github.com/flutter/plugins/blob/b548ae18646d1725a8e1f004c26bf42e4d362aa8/packages/google_sign_in/google_sign_in/ios/Classes/FLTGoogleSignInPlugin.m#L246-L250
which means I'd have to mock out the delegate calls.  
GoogleSignIn 6.0 (https://github.com/flutter/flutter/issues/86436) removed this delegate and made it a `disconnectWithCallback:` so the above code will become:
```obj-c
    [GIDSignIn.sharedInstance disconnectWithCallback:^(NSError * _Nullable error) {
      [self respondWithAccount:@{} result:result error:nil];
    }];
```
which will be super easy to test.

TL;DR I started to write tests for these results, but it would have been a lot of throwaway work for the next review.  Instead, I wrote all the tests that seemed like the best bang for my buck, and the next review tests diff will be easier to read, and not a wall o'green like this review.

Reply:
Makes sense, thanks!